src/face_registry.cc

```diff
@@ -42,6 +42,42 @@ static Face parse_face(StringView facedesc)
     return res;
 }
 
+String attributes_to_str(Attribute attributes)
+{
+    if (attributes == Attribute::Normal)
+        return "";
+
+    struct Attr { Attribute attr; StringView name; }
+    attrs[] {
+        { Attribute::Exclusive, "e" },
+        { Attribute::Underline, "u" },
+        { Attribute::Reverse, "r" },
+        { Attribute::Blink, "B" },
+        { Attribute::Bold, "b" },
+        { Attribute::Dim, "d" },
+        { Attribute::Italic, "i" },
+    };
+
+    auto filteredAttrs = attrs |
+                         filter([=](const Attr& a) { return attributes & a.attr; }) |
+                         transform([](const Attr& a) { return a.name; });
+    String res;
```

Comment:
`return accumulate(filteredAttrs, String{"+"}, std::plus<>{});` should do the trick here.